src/services/encryptionService.js

```diff
@@ -1,38 +1,170 @@
+      const encryptedKey = this.#account.pickle(key);
+      localforage.setItem("account", encryptedKey);
+    }
+    return this.#account;
   }
 
-  async registerDevice() {
-    const user = this.#getAccount();
-    user.generate_one_time_keys(50);
+  async logout() {
+    this.#account = null;
+  }
+
+  async registerDevice(password) {
+    const user = await this.#getAccount(password);
+    if (!user) return;
 
-    setEncryptedUser(user);
+    user.generate_one_time_keys(50);
+    console.log("Encryption: User account generated", user);
 
     const data = {
       identity_key: user.curve25519_key,
       signed_key: user.ed25519_key,
       one_time_pre_keys: Object.fromEntries(user.one_time_keys.entries()),
     };
-    await api.encryptedDeviceCreate(data);
+
+    try {
+      await api.encryptedDeviceCreate(data);
+      if (this.#chatIdAfterRegister) {
+        navigateTo(`/#${this.#chatIdAfterRegister}`);
+        store.dispatch(
+          setSelectedConversation({ id: this.#chatIdAfterRegister })
+        );
+      }
+    } catch (error) {
+      console.error("[encryption] Failed to register device", error);
+    }
+  }
+
+  async createEncryptedChat(userId, userObject) {
+    console.log("Encryprion: createEncryptedChat ");
+
+    if (!userId) {
+      console.error("[encryption] User ID is required to create a chat");
+      return;
+    }
+
+    const requestData = {
+      type: "u",
+      participants: [userId],
+      is_encrypted: true,
+    };
+
+    try {
+      const chat = await api.conversationCreate(requestData);
+      userObject && store.dispatch(addUsers([userObject]));
+      store.dispatch(insertChat({ ...chat, messagesIds: null }));
+      return chat._id;
+    } catch (error) {
+      console.error("[encryption] Failed to create encrypted chat", error);
+      return null;
+    }
+  }
+
+  async #getParticipantsKeys(userId) {
+    try {
+      const usersKeys = await api.getEncryptedKeys({ user_ids: [userId] });
+      const keys = usersKeys[userId]?.[0];
+
+      if (keys) {
+        store.dispatch(upsertUser({ _id: userId, keys }));
+      } else {
+        console.error("[encryption] No keys found for the user", userId);
+      }
+
+      return keys;
+    } catch (error) {
+      console.error("[encryption] Failed to get participant's keys", error);
+      return null;
+    }
+  }
+
+  async createEncryptionSession(userId) {
+    if (!this.#vodozemacInitialized || !this.#account) {
+      console.error("[encryption] Service or account not initialized");
+      return null;
+    }
+
+    const userKeys = await this.#getParticipantsKeys(userId);
+    if (!userKeys) {
+      console.error("[encryption] Could not retrieve opponent's keys");
+      return null;
+    }
+
+    try {
+      const session = this.#account.create_outbound_session(
+        userKeys.identity_key,
+        userKeys.one_time_pre_keys
+      );
+      this.#encryptionSession = session;
+      store.dispatch(
```

Comment:
this also could moved outside here, which looks like a side effect.
If a function called `createEncryptionSession ` it should just create and return a session